infra/monitoring.tf

```diff
@@ -0,0 +1,88 @@
+# Creates a metric that counts the number of log entries containing 'Relay succeeded' in the relay cloud function.
+resource "google_logging_metric" "successful_relay_count" {
+  project     = module.oracle_relayer.project_id
+  name        = "successful_relay_count"
+  description = "Number of log entries containing 'Relay succeeded' in the relay cloud function"
+  filter      = <<EOF
+    severity>=DEFAULT
+    SEARCH("`Relay succeeded`")
+    resource.labels.function_name="${google_cloudfunctions2_function.relay.name}"
+  EOF
+
+  metric_descriptor {
+    metric_kind = "DELTA"
+    value_type  = "INT64"
+    unit        = "1"
+
+    labels {
+      key = "ratefeed"
+    }
+  }
+
+  label_extractors = {
+    "ratefeed" = "EXTRACT(labels.rateFeed)"
+  }
+}
+
+# Discord notification channel for info-only alerts
+resource "google_monitoring_notification_channel" "discord_channel" {
+  project      = module.oracle_relayer.project_id
+  display_name = "Discord #relayer-alerts"
+  type         = "webhook_tokenauth"
+
+  labels = {
+    url = terraform.workspace == "prod" ? var.discord_webhook_url_prod : var.discord_webhook_url_staging
+  }
+}
+
+# Splunk notification channel for on-call alerts
+resource "google_monitoring_notification_channel" "victorops_channel" {
+  project      = module.oracle_relayer.project_id
+  display_name = "Splunk (VictorOps)"
+  type         = "webhook_tokenauth"
+
+  labels = {
+    url = var.victorops_webhook_url
+  }
+}
+
+# Creates an alert policy that triggers when no successful relay logs have been received in the last 30 minutes,
+# and sends a notification to Splunk.
+resource "google_monitoring_alert_policy" "successful_relay_policy" {
+  project      = module.oracle_relayer.project_id
+  display_name = "no-successful-relay-logs"
+  combiner     = "OR"
```

Comment:
From the terraform docs:

> [combiner](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/monitoring_alert_policy#combiner) - (Required) How to combine the results of multiple conditions to determine if an incident should be opened. Possible values are: AND, OR, AND_WITH_MATCHING_RESOURCE.

although I think it has no effect in our case since it's a single condition

Reply:
yeah, that was my thought as well, why do we need "OR" if there's no second part to the condition

Reply:
Yeah it's strange. It's also the default combiner that you get when you create a policy through the UI even if it's only a single condition, so at least this is consistent with that

Reply:
And does it still work if you remove this line?

Reply:
nope, it's required by the tf definition (also the linter complaints when you remove it). I added a comment to clarify that's not used in practice though